src/main/java/net/sf/jabref/gui/ParserResultWarningDialog.java

```
@@ -0,0 +1,90 @@
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.gui;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.swing.JOptionPane;
+
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.logic.l10n.Localization;
+
+public class ParserResultWarningDialog {
+
+    /**
+     * Shows a dialog with the warnings from an import or open of a file
+     *
+     * @param parserResult - ParserResult for the current import/open
+     * @param jabRefFrame - the JabRefFrame
+     */
+    public static void showParserResultWarningDialog(ParserResult parserResult, JabRefFrame jabRefFrame) {
+        Objects.requireNonNull(parserResult);
+        Objects.requireNonNull(jabRefFrame);
+        showParserResultWarningDialog(parserResult, jabRefFrame, Integer.MAX_VALUE, -1);
+    }
+
+    /**
+     * Shows a dialog with the warnings from an import or open of a file
+     *
+     * @param parserResult - ParserResult for the current import/open
+     * @param jabRefFrame - the JabRefFrame
+     * @param maxWarnings - Maximum number of warnings to display
+     * @param dataBaseNumber - Database tab number to activate when showing the warning dialog
+     */
+    public static void showParserResultWarningDialog(ParserResult parserResult, JabRefFrame jabRefFrame,
+            int maxWarnings, int dataBaseNumber) {
+        Objects.requireNonNull(parserResult);
+        Objects.requireNonNull(jabRefFrame);
+        if (parserResult.hasWarnings()) {
```

I would again use fail fast, namely, return quickly if there are no warnings in this method.